src/libmongoc/src/mongoc/mongoc-topology-description-apm.c

```diff
@@ -115,10 +114,34 @@ _mongoc_topology_description_monitor_opening (mongoc_topology_description_t *td)
    for (i = 0; i < td->servers->items_len; i++) {
       sd = (mongoc_server_description_t *) mongoc_set_get_item (td->servers,
                                                                 (int) i);
-
       _mongoc_topology_description_monitor_server_opening (td, sd);
    }
 
+   /* If this is a load balanced topology:
+    * - update the one server description to be LoadBalancer
+    * - emit a server changed event Unknown => LoadBalancer
+    * - emit a topology changed event
+    */
+   if (td->type == MONGOC_TOPOLOGY_LOAD_BALANCED) {
+      mongoc_server_description_t *prev_sd;
+
+      /* LoadBalanced deployments must have exactly one host listed. Otherwise,
+       * an error would have occurred when constructing the topology. */
+      BSON_ASSERT (td->servers->items_len == 1);
+      sd = (mongoc_server_description_t *) mongoc_set_get_item (td->servers, 0);
```

Comment:
Since when are sets indexed 😆 ; I guess this just gets the one server description?

Reply:
Haha, that is an idiosyncrasy of `mongoc_set_t`. But this does indeed get the one and only item in the set.